sds/src/scanner/mod.rs

```diff
@@ -455,11 +461,12 @@ impl Scanner {
 
         // Refill the rule_matches with the validated matches
         for (_, mut matches) in match_validator_rule_match_per_type {
-            rule_matches.append(&mut matches);
+            validated_rule_matches.append(&mut matches);
         }
 
         // Sort rule_matches by start index
-        rule_matches.sort_by_key(|rule_match| rule_match.start_index);
+        validated_rule_matches.sort_by_key(|rule_match| rule_match.start_index);
+        *rule_matches = validated_rule_matches;
```

Comment:
I don't think it's related to this PR but the ordering is different from this [one](https://github.com/DataDog/dd-sensitive-data-scanner/blob/e50ab079ab160ae97f5c2d512641ea16fd575dff/sds/src/scanner/mod.rs#L582-L601). It may be something to align in a follow-up PR to avoid two ordering function

Reply:
I think this is fine because all overlapping matches have already been removed. Most of the complication is dealing with overlapping matches.

Reply:
I agree that here it's not necessarily impacting our code. However, it looks like a unexpected behavior for users of the library. Someone integrating the library may assume that this method does not impact the order of matches but only alter the `match_status`. I think not alerting the order would make the use of this code less error-prone. The sorting at the end also make the time complexity to at least `O(n*log(n))` where `n` is the number of matches.

Instead of re-ordering the matches, would it be possible to keep the order in the method as so avoid the sorting and avoid the possible re-ordering.